api/repositories/service_binding_repository_test.go

```diff
@@ -237,7 +243,103 @@ var _ = Describe("ServiceBindingRepo", func() {
+
+			space2 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space2"))
+			cfApp2 := createAppCR(testCtx, k8sClient, "app-2-name", generateGUID(), space2.Name, "STOPPED")
+			serviceInstance2GUID = generateGUID()
+			cfServiceInstance2 := createServiceInstanceCR(testCtx, k8sClient, serviceInstance2GUID, space2.Name, "service-instance-2-name", "secret-2-name")
+			serviceBinding2 = createServiceBindingCR(testCtx, k8sClient, generateGUID(), space2.Name, nil, cfServiceInstance2.Name, cfApp2.Name)
+
+			requestMessage = repositories.ListServiceBindingsMessage{}
+		})
+
+		When("the user has access to both namespaces", func() {
+			BeforeEach(func() {
+				createRoleBinding(testCtx, userName, spaceDeveloperClusterRole.Name, space.Name)
+				createRoleBinding(testCtx, userName, spaceDeveloperClusterRole.Name, space2.Name)
+			})
+
+			Describe("no query parameters are specified", func() {
+				It("returns a list of ServiceBindingRecords in the spaces the user has access to", func() {
+					Eventually(func() []repositories.ServiceBindingRecord {
+						responseServiceBindings, err := repo.ListServiceBindings(context.Background(), authInfo, requestMessage)
+						Expect(err).NotTo(HaveOccurred())
+						return responseServiceBindings
+					}, timeCheckThreshold*time.Second).Should(ConsistOf(
+						MatchFields(IgnoreExtras, Fields{
+							"GUID":                Equal(serviceBinding1.Name),
+							"Type":                Equal("app"),
+							"Name":                Equal(serviceBinding1.Spec.Name),
+							"AppGUID":             Equal(serviceBinding1.Spec.AppRef.Name),
+							"ServiceInstanceGUID": Equal(serviceBinding1.Spec.Service.Name),
+							"SpaceGUID":           Equal(serviceBinding1.Namespace),
+							"LastOperation": MatchFields(IgnoreExtras, Fields{
+								"Type":  Equal("create"),
+								"State": Equal("succeeded"),
+							}),
+						}),
+						MatchFields(IgnoreExtras, Fields{
+							"GUID":                Equal(serviceBinding2.Name),
+							"Type":                Equal("app"),
+							"Name":                Equal(serviceBinding2.Spec.Name),
+							"AppGUID":             Equal(serviceBinding2.Spec.AppRef.Name),
+							"ServiceInstanceGUID": Equal(serviceBinding2.Spec.Service.Name),
+							"SpaceGUID":           Equal(serviceBinding2.Namespace),
+							"LastOperation": MatchFields(IgnoreExtras, Fields{
+								"Type":  Equal("create"),
+								"State": Equal("succeeded"),
+							}),
+						}),
+					))
+				})
+			})
+
+			When("query parameters are specified", func() {
+				BeforeEach(func() {
+					requestMessage = repositories.ListServiceBindingsMessage{
+						ServiceInstanceGUIDs: []string{serviceInstance2GUID},
+					}
+				})
+
+				It("eventually returns only a subset of the ServiceBindings that match the provided filter", func() {
+					Eventually(func() []repositories.ServiceBindingRecord {
+						responseServiceBindings, err := repo.ListServiceBindings(context.Background(), authInfo, requestMessage)
```

Comment:
`requestMessage` is a local to this invocation, so could be inlined, instead of initialising it in a `BeforeEach` above. Alternatively, the `ListServiceBindings` function could be invoked in `JustBeforeEach` so that it is not invoked in `Its`. Then, setting up the request message in `BeforeEach` above would make sense.